lib/src/lifecycle_module.dart

```diff
@@ -889,6 +889,17 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
     }
   }
 
+  @mustCallSuper
```

Comment:
So that consumers who override `onDispose` and don't already call super, should we put this cleanup logic somewhere else?

Perhaps in the constructor:
```dart
getManagedDisposer(() async {
  if (isInstantiated || isUnloaded) {
    return;
  }
  _childModules.clear();
});